src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -479,12 +480,13 @@ transitiveClosure = coerce AM.transitiveClosure
 -- @
 -- topSort 'empty'                 == []
 -- topSort ('vertex' x)            == [x]
--- topSort (1 * (2 + 4) + 3 * 4) == [3, 1, 4, 2]
+-- topSort (1 * (2 + 4) + 3 * 4) == [1, 2, 3, 4]
 -- topSort ('join' x y)            == 'fmap' 'Left' (topSort x) ++ 'fmap' 'Right' (topSort y)
--- topSort                       == 'Data.Maybe.fromJust' . topSort . 'fromAcyclic'
+-- 'Right' . topSort               == 'AM.topSort' . 'fromAcyclic'
 -- @
-topSort :: (Ord a) => AdjacencyMap a -> [a]
-topSort (AAM am) = Typed.topSort (Typed.fromAdjacencyMap am)
+topSort :: Ord a => AdjacencyMap a -> [a]
+topSort (AM.topSort . coerce -> Right vs) = vs
```

Comment:
I find this rather cryptic: why not do a simple pattern matching on the result of `AM.topSort`?

Reply:
(I guess I'm just not a fan of `ViewPatterns`, although in some cases its use is justified.)

Reply:
Yeah, I probably went overboard with ViewPatterns, I think most can be removed